File: everyDay/2325.解密消息.cpp
/*
 * @lc app=leetcode.cn id=2325 lang=cpp
 *
 * [2325] 解密消息
 */

// @lc code=start
class Solution
{
public:
    string decodeMessage(string key, string message) {
        string ans;
        int n = message.size();
        char s[26]{0};
        for (int i = 0, c = 0; i < key.size(); i++) {
            if (key[i] != ' ' && !s[key[i] - 'a']) {
                s[key[i] - 'a'] = c + 'a';
                c++;
            }
        }
        for (int i = 0; i < n; i++) {
            if (message[i] != ' ') {
                ans += s[message[i] - 'a'];
            } else
                ans += ' ';
        }
        return ans;
    }
};
// @lc code=end

File: everyDay/2325.解密消息.go
/*
 * @lc app=leetcode.cn id=2325 lang=golang
 *
 * [2325] 解密消息
 */

// @lc code=start
func decodeMessage(key string, message string) (string) {
    cur := byte('a')
    rules := map[rune]byte{}

    for _, c := range key {
        if c != ' ' && rules[c] == 0 {
            rules[c] = cur
            cur++
        }
    }
	m := []byte(message)
    for i, c := range message {
        if c != ' ' {
            m[i] = rules[c]
        }
    }

    return string(m)
}
// @lc code=end

